### DISCUSSION TITLE
How to disable titles?

### DISCUSSION THREAD

**Discussion by TVBZ** [thephpleague/commonmark#907] [Q&A] [answered]
New to commonmark, using it in Laravel project, inside livewire component. 
I can't seam to find anything in documentation on how to disable `<h1>`, `<h2>`,... title tags in config file.
There's an option to disable bold, italic and so on. But I want to avoid my users using title tags.
In best case scenario I would allow 1 level title tag, for example` # my title`, but it would have to be rendered as a `<h3>` tag. 
But I would be satisfied if I can just disable titles all together.

Is it possible to configure to **disable title tags completely?** 
Or is it possible to enable 1 level and configure how it would be rendered?

Thanks.

**Comment by colinodell** [comment, marked as answer]
There are two ways you can solve this.

One way is to create a custom [event listener](https://commonmark.thephpleague.com/2.3/customization/event-dispatcher/#registering-listeners) (sorta [like this](https://commonmark.thephpleague.com/2.3/customization/event-dispatcher/#example)) that will [iterate the parsed elements](https://commonmark.thephpleague.com/2.3/customization/abstract-syntax-tree/), locate the headings, and do _something_ with them.  You could potentially remove them (by calling `->detach()`), [change the heading level](https://github.com/thephpleague/commonmark/blob/2.3/src/Extension/CommonMark/Node/Block/Heading.php#L37), or replace it with a `Paragraph` node.  This would be considered the "best practice" way of doing things.

The other solution is a little bit more of a hack, but would also work just fine.  Basically, replace the existing [HeadingRenderer](https://github.com/thephpleague/commonmark/blob/2.3/src/Extension/CommonMark/Renderer/Block/HeadingRenderer.php) with [your own custom one](https://commonmark.thephpleague.com/2.3/customization/rendering/) that renders different HTML output.

Hope that helps!

**Comment by TVBZ** [comment]
Thanks @colinodell, I will dig into event listeners
